src/libraries/System.Collections.Concurrent/src/System/Collections/Concurrent/ConcurrentDictionary.cs

```diff
@@ -34,62 +22,56 @@ namespace System.Collections.Concurrent
+        /// <summary>Whether to dynamically increase the size of the striped lock.</summary>
+        private readonly bool _growLockArray;
+        /// <summary>The maximum number of elements per lock before a resize operation is triggered.</summary>
+        private int _budget;
 
-        // The default capacity, i.e. the initial # of buckets. When choosing this value, we are making
-        // a trade-off between the size of a very small dictionary, and the number of resizes when
-        // constructing a large dictionary. Also, the capacity should not be divisible by a small prime.
+        /// <summary>The default capacity, i.e. the initial # of buckets.</summary>
+        /// <remarks>
+        /// When choosing this value, we are making a trade-off between the size of a very small dictionary,
+        /// and the number of resizes when constructing a large dictionary. Also, the capacity should not be
+        /// divisible by a small prime.
+        /// </remarks>
         private const int DefaultCapacity = 31;
 
-        // The maximum size of the striped lock that will not be exceeded when locks are automatically
-        // added as the dictionary grows. However, the user is allowed to exceed this limit by passing
-        // a concurrency level larger than MaxLockNumber into the constructor.
+        /// <summary>
+        /// The maximum size of the striped lock that will not be exceeded when locks are automatically
+        /// added as the dictionary grows.
+        /// </summary>
+        /// <remarks>
+        /// The user is allowed to exceed this limit by passing
+        /// a concurrency level larger than MaxLockNumber into the constructor.
+        /// </remarks>
         private const int MaxLockNumber = 1024;
 
-        // Whether TValue is a type that can be written atomically (i.e., with no danger of torn reads)
+        /// <summary>Whether TValue is a type that can be written atomically (i.e., with no danger of torn reads).</summary>
         private static readonly bool s_isValueWriteAtomic = IsValueWriteAtomic();
 
-        /// <summary>
-        /// Determines whether type TValue can be written atomically
-        /// </summary>
+        /// <summary>Determines whether type TValue can be written atomically.</summary>
         private static bool IsValueWriteAtomic()
         {
-            //
             // Section 12.6.6 of ECMA CLI explains which types can be read and written atomically without
-            // the risk of tearing.
-            //
-            // See http://www.ecma-international.org/publications/files/ECMA-ST/Ecma-335.pdf
-            //
-            Type valueType = typeof(TValue);
-            if (!valueType.IsValueType)
+            // the risk of tearing. See https://www.ecma-international.org/publications/files/ECMA-ST/ECMA-335.pdf
+
+            if (!typeof(TValue).IsValueType ||
+                typeof(TValue) == typeof(IntPtr) ||
```

Comment:
Aren't `IntPtr`/`UIntPtr`  `IsValueType == true`? Is it because they are unspecified size?

Reply:
> Aren't IntPtr/UIntPtr IsValueType == true?

Yes.  This is saying we can return true immediately if it's a reference type or if it's {U}IntPtr.

Reply:
Ah, missed the `!`